### ISSUE TITLE
TOO_MANY_REGISTRATIONS FCM crash from pinpoint 

### ISSUE THREAD

**Issue by jeffville** [aws-amplify/amplify-android#2555]
### Before opening, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-android/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-android/discussions).

### Language and Async Model

Kotlin

### Amplify Categories

Notifications

### Gradle script dependencies

<details>

```groovy
// Put output below this line
implementation "com.amplifyframework:aws-api:2.11.1"
implementation "com.amplifyframework:aws-auth-cognito:2.11.1"
implementation "com.amplifyframework:aws-push-notifications-pinpoint:2.11.1"

```

</details>

### Environment information

<details>

```
# Put output below this line

------------------------------------------------------------
Gradle 7.5
------------------------------------------------------------
Kotlin:       1.6.21
Groovy:       3.0.10

```

</details>

### Describe the bug

We are getting a crash FCM: TOO_MANY_REGISTRATIONS on Pinpoint SDK (AWSPinpointPushNotificationsPlugin.kt:167) when we submit the app in Playstore internal testing track

`xception com.google.android.gms.tasks.RuntimeExecutionException: java.io.IOException: java.util.concurrent.ExecutionException: java.io.IOException: TOO_MANY_REGISTRATIONS
  at com.google.android.gms.tasks.zzw.getResult (com.google.android.gms:play-services-tasks@@18.0.2:3)
  at com.amplifyframework.pushnotifications.pinpoint.AWSPinpointPushNotificationsPlugin.fetchFCMDeviceToken$lambda-4 (AWSPinpointPushNotificationsPlugin.kt:167)
  at androidx.navigation.ui.NavigationUI$$ExternalSyntheticLambda0.b (R8$$SyntheticClass:11)
  at com.google.android.gms.tasks.zzi.run (zzi.java:1)
  at android.os.Handler.handleCallback (Handler.java:942)
  at android.os.Handler.dispatchMessage (Handler.java:99)
  at android.os.Looper.loopOnce (Looper.java:201)
  at android.os.Looper.loop (Looper.java:288)
  at android.app.ActivityThread.main (ActivityThread.java:7872)
  at java.lang.reflect.Method.invoke
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:548)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:936)
Caused by java.io.IOException: java.util.concurrent.ExecutionException: java.io.IOException: TOO_MANY_REGISTRATIONS`

I think this is due to the devices used on the automated testing by Google when they run the pre-launch report but shouldn't the crash be handle on the Pinpoint SDK rather than throwing the exception?

### Code Snippet

```java
// Put your code below this line.

```

### Log output

<details>

```
// Put your logs below this line


```

</details>

### GraphQL Schema

<details>

```graphql
// Put your schema below this line


```

</details>

**Comment by tylerjroach** [comment]
Are you seeing this as a fatal or a non-fatal exception? 

We already wrap `com.google.android.gms.tasks.zzw.getResult` in a try/catch IOException. Is it possible this is just getting logged (but not a fatal crash)?

You are correct that this appears to be common in pre-launch reporting but in normal user scenarios, this should not be an issue.

**Comment by jeffville** [comment]
Hi @tylerjroach It was being returned as a fatal crash both captured by google pre-launch reports and crashlytics.

**Comment by tylerjroach** [comment]
@jeffville Thank you for the information. we have the getResult block in a try/catch already. If it is not being caught, it may be throwing within. an uncatchable thread and out of our control on Amplify. Please let us know if you see this crash impacting customers, outside of the pre-launch report.

**Comment by jeffville** [comment]
Hi @tylerjroach thanks for the response, yes upon our release we saw crash impacting our customers followed by the stacktrace on this ticket:  https://github.com/aws-amplify/amplify-android/issues/2559

**Comment by tjleing** [comment]
Looking into it a bit more, it looks like the error may be a `RuntimeExecutionException` which wraps an `IOException`, so our try/catch isn't actually detecting it.  I'm creating a PR to address this which would also address #2559 for the same reason, but @jeffville do you have a way to reproduce either issue other than submitting apps to the playstore so that we can make sure this fixes the issue?

**Comment by jeffville** [comment]
Hi @tjleing thanks for the submitting PR fix for the issue. We're trying to reproduce it locally but unfortunately we aren't able to do so other than submitting the apps.

**Comment by tjleing** [comment]
Sorry to hear that -- once this PR is out it should be released next week!

**Comment by jeffville** [comment]
Thanks @tjleing will update once the new version is released and keep you posted if we still encounter the issue

**Comment by gpanshu** [comment]
Hi @jeffville you can use version [2.12.0](https://github.com/aws-amplify/amplify-android/releases/tag/release_v2.12.0) which has this fix or the latest which is [2.13.0](https://github.com/aws-amplify/amplify-android/releases/tag/release_v2.13.0). Thank you.
